plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -727,17 +724,18 @@ func PodFitsHostPorts(pod *api.Pod, meta interface{}, nodeInfo *schedulercache.N
 		wantPorts = getUsedPorts(pod)
 	}
 	if len(wantPorts) == 0 {
-		return true, nil
+		return true, nil, nil
 	}
 
 	// TODO: Aggregate it at the NodeInfo level.
 	existingPorts := getUsedPorts(nodeInfo.Pods()...)
+	var predicateFails []algorithm.PredicateFailureReason
 	for wport := range wantPorts {
 		if wport != 0 && existingPorts[wport] {
-			return false, ErrPodNotFitsHostPorts
+			predicateFails = append(predicateFails, ErrPodNotFitsHostPorts)
```

Comment:
We were once discussing extending the errors with some optional additional information - this way it would be a good place to put the info about the port and then return all the errors.